### PR TITLE
#vp-986 SendGrid fix

### PR DESCRIPTION
**Pull request by kostyrin**

### CODE REVIEW

File: src/VirtoCommerce.NotificationsModule.Core/Model/Notification.cs [hunk 1]
```diff
@@ -10,14 +9,15 @@ namespace VirtoCommerce.NotificationsModule.Core.Model
...
-    public abstract class Notification : AuditableEntity, ICloneable
+    public abstract class Notification : AuditableEntity, ICloneable, IHasOuterId
     {
         /// <summary>
         /// For detecting owner
         /// </summary>
         public TenantIdentity TenantIdentity { get; set; } = TenantIdentity.Empty;
         public bool IsActive { get; set; }
         public string LanguageCode { get; set; }
+        public string Alias { get; set; }
```
Comment (tatarincev) [inline src/VirtoCommerce.NotificationsModule.Core/Model/Notification.cs]: Add comments for this property. e.g `This field represents an alias for the notification type and is used only for backward compatibility with old notification names that are stored and used by API clients.`

File: src/VirtoCommerce.NotificationsModule.Core/Types/ConfirmationEmailNotification.cs [hunk 2]
```diff
@@ -4,8 +4,11 @@ namespace VirtoCommerce.NotificationsModule.Core.Types
 {
     public class ConfirmationEmailNotification : EmailNotification
     {
-        //need to the storefront
-        public override string Type { get; set; } = "EmailConfirmationNotification";
+        public ConfirmationEmailNotification()
+        {
+            Alias = "EmailConfirmationNotification";
```
Comment (tatarincev) [inline src/VirtoCommerce.NotificationsModule.Core/Types/ConfirmationEmailNotification.cs]: Add comment `for backward compatibility`

File: src/VirtoCommerce.NotificationsModule.Data/Services/NotificationSearchService.cs [hunk 3]
```diff
@@ -25,68 +25,86 @@ public NotificationSearchService(Func<INotificationRepository> repositoryFactory
...
-            }
             return result;
         }
 
         protected virtual IQueryable<NotificationEntity> BuildQuery(INotificationRepository repository, NotificationSearchCriteria criteria, IEnumerable<SortInfo> sortInfos)
         {
             var query = repository.Notifications;
 
-            if (!string.IsNullOrEmpty(criteria.NotificationType))
+            if (!criteria.NotificationTypes.IsNullOrEmpty())
```
Comment (tatarincev) [inline src/VirtoCommerce.NotificationsModule.Data/Services/NotificationSearchService.cs]: The notification "Alias" can be checked here

File: src/VirtoCommerce.NotificationsModule.Data/Services/NotificationSearchService.cs [hunk 4]
```diff
@@ -25,68 +25,86 @@ public NotificationSearchService(Func<INotificationRepository> repositoryFactory
...
                 if (criteria.Take > 0)
                 {
-                    var notificationIds = await query.OrderBySortInfos(sortInfos).ThenBy(x=>x.Id)
+                    var notificationIds = await query.OrderBySortInfos(sortInfos).ThenBy(x => x.Id)
                                                      .Select(x => x.Id)
                                                      .Skip(criteria.Skip).Take(criteria.Take)
                                                      .ToArrayAsync();
                     var unorderedResults = await _notificationService.GetByIdsAsync(notificationIds, criteria.ResponseGroup);
-                    result.Results = unorderedResults.OrderBy(x => Array.IndexOf(notificationIds, x.Id)).ToList();
+                    result = unorderedResults.OrderBy(x => Array.IndexOf(notificationIds, x.Id)).ToArray();
```
Comment (tatarincev) [inline src/VirtoCommerce.NotificationsModule.Data/Services/NotificationSearchService.cs]: Seems you have broken multiple data source pagination with transient and persisted notifications.
Could you return back the original code?
In additional need to apply to `ReduceDetails(criteria.ResponseGroup);` for all notifications not only for transient as it made before.

File: src/VirtoCommerce.NotificationsModule.Data/Services/NotificationSearchService.cs (hunk 3 above)
Comment (kostyrin) [reply src/VirtoCommerce.NotificationsModule.Data/Services/NotificationSearchService.cs]: then need to add the property to DB(add migration and etc..), is it OK?

Comment (kostyrin) [reply src/VirtoCommerce.NotificationsModule.Data/Services/NotificationSearchService.cs]: look at the code, make some refactor

**Pull request merged**